### PR TITLE
@damassi => Location autocomplete and initial connection to convection

### PR DESCRIPTION
**Pull request by sweir27**
This PR adds two things:
- Location autocomplete based on Google's places API (we use this around Force already, and it's what powers the location-based autocomplete in the current system)
  - I configured it to only return results at the `city` level, and from a subsequent `getDetails()` call we are able to get fine-grained data on the city, state, and country.
- Hookup to convection (woo!)
  - After clicking "Next" on the "Describe your work" section, we save a submission in Convection. On the next page (currently blank except for a submit button... but this is where you'll upload photos), we send a request with `state: 'submitted'`. This tells convection that you are ready to submit the consignment, and we send you/artsy specialists an email. It all works locally with emails going to mailtrap. :D 

Note that it's still in the early stages, so things like error handling is not super fleshed out. In the next PR I intend to clean up some of the loose edges and add an error state so you know if your submission was not accepted for some reason. 

Flow:
flow-with-loc

Proof of email:
image

### CODE REVIEW

File: desktop/apps/consignments2/client/actions.js [hunk 1]
```diff
@@ -21,32 +26,107 @@ export function chooseArtistAndAdvance (value) {
...
-    // TODO: async request to convection to create submission
-    dispatch(updateSubmission()) // update state
-    dispatch(incrementStep()) // move to next step
+  return async (dispatch, getState) => {
+    try {
+      const {
+        body: {
+          token
+        }
+      } = await request
```
Comment (sweir27) [inline desktop/apps/consignments2/client/actions.js]: We have to get a JWT from gravity first, which is encoded using convection's `internal_secret`. We then use this to auth with convection and create a submission for the right `user_id`.

File: desktop/apps/consignments2/client/actions.js [hunk 2]
```diff
@@ -21,32 +26,107 @@ export function chooseArtistAndAdvance (value) {
...
+        const countryDisplay = country && country.long_name
+        const stateDisplay = state && state.long_name
+
+        dispatch(updateLocationInputValues(cityDisplay, stateDisplay, countryDisplay))
+      }
+    }
+
+    try {
+      if (window.google) {
+        const placesService = new window.google.maps.places.PlacesService(document.createElement('div'))
```
Comment (damassi) [inline desktop/apps/consignments2/client/actions.js]: Weird google api lol `createElement`

Comment (sweir27) [inline desktop/apps/consignments2/client/actions.js]: oh ugh. yeah you have to either pass in a full-fledged `map` object there, or a dummy node on the page 😞

File: desktop/apps/consignments2/components/describe_work/index.js [hunk 3]
```diff
@@ -35,16 +40,45 @@ function validate (values) {
...
+
+function renderSuggestion (suggestion) {
+  return (
+    <div className='autosuggest-suggestion'>
+      <div>{suggestion.description}</div>
+    </div>
+  )
+}
+
 let DescribeWork = props => {
```
Comment (damassi) [inline desktop/apps/consignments2/components/describe_work/index.js]: `const` or maybe `function DescribeWork` for consistency?

**Comment by damassi** [comment]
clarity-of-thought

Will merge on 🍏

**Pull request merged**